### PR TITLE
Add quilt-cli package

### PR DESCRIPTION
**Pull request by bentsherman**
This PR adds the quilt cli wrapper into a subdirectory. When the PR is merged, quilt-cli can be installed via pip:
```bash
pip3 install git+https://github.com/nextflow-io/nf-quilt.git#subdirectory=quilt-cli
```

Additionally, https://github.com/seqeralabs/quilt-api can be deleted, as it has been transferred in its entirety to this repo.

### CODE REVIEW

**Comment by bentsherman** [comment]
I'm looking at the possibility of compiling the python code into a standalone executable with Pyinstaller:

https://pyinstaller.org/en/stable/

I haven't gotten it to work yet, but it might allow us to bypass the pip install process. Is it possible to package executables like this into a JAR?

File: quilt-cli/quilt_cli/__main__.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python3
+# Copyright (C) Seqera Labs S.L.- All Rights Reserved
+# Unauthorized copying of this file, via any medium is strictly prohibited
+# Proprietary and confidential
+# Written by Harshil Patel <harshil.patel@seqera.io>
+#            Ben Sherman <ben.sherman@seqera.io>
+#            Graham Wright <graham.wright@seqera.io>
```
Comment (pditommaso) [inline quilt-cli/quilt_cli/__main__.py:18]: The usual Apache 2.0 license boilerplate should be used

Comment (bentsherman) [reply quilt-cli/quilt_cli/__main__.py:18]: Done

**Comment by bentsherman** [comment]
I don't think this pyinstaller thing is going to work for us. So this PR should be good to merge now.

Also we need to create a new release so that Harshil can start testing it with Tower.

**Review by pditommaso — APPROVED** [review]
Ok

**Comment by bentsherman** [comment]
Can you make a new release as well?

**Comment by pditommaso** [comment]
Let me do it

**Comment by pditommaso** [comment]
is there any change in nextflow requirement, right? `Plugin-Requires: >=22.04.0`

**Comment by bentsherman** [comment]
I don't think so, the plugin only uses the basic observer hooks -- `onFlowCreate`, `onFilePublish`, `onFlowComplete`

**Comment by pditommaso** [comment]
Done https://github.com/nextflow-io/plugins/commit/d57589ddad461ad2c37c157c6f5843cbf79a91dd

**Pull request merged**